packages/xo-web/src/common/xo/index.js

```diff
@@ -1591,41 +1601,45 @@ export const deleteTemplates = templates =>
     )
 
     const nDefaultTemplates = defaultTemplates.length
-    if (nDefaultTemplates === 0 && nErrors === 0) {
-      return
-    }
+    const nProtectedTemplates = protectedTemplates.length
 
-    const showError = () =>
-      error(
-        _('failedToDeleteTemplatesTitle', { nTemplates: nErrors }),
-        _('failedToDeleteTemplatesMessage', { nTemplates: nErrors })
+    if (nProtectedTemplates !== 0) {
+      await confirm({
+        title: _('deleteProtectedTemplatesTitle', { nProtectedTemplates }),
+        body: _('deleteProtectedTemplatesMessage', { nProtectedTemplates }),
+      })
+      await Promise.all(
+        map(protectedTemplates, id =>
+          _call('vm.delete', {
+            id,
+            forceBlockedOperation: true,
+          }).catch(() => {
+            nErrors++
```

Comment:
As [@MathieuRA mentioned](https://github.com/vatesfr/xen-orchestra/pull/7493#pullrequestreview-1955055689), I don't think anything is preventing a template from being both a default template a protected against deletion. So here, instead of immediately counting it as an error, we should first test if the error is because it's a default template, and if it is, we should add it to the `defaultTemplates` to later ask the user if they want to force the deletion for that reason too.

For that to work, we should also remember the user's decision regarding protected templates and then, in the 2nd call, pass the `forceBlockedOperation` option with the same value.